src/Features/VisualBasic/Portable/Completion/VisualBasicCompletionService.vb

```diff
@@ -151,18 +149,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion
             Return CompletionUtilities.GetCompletionItemSpan(text, caretPosition)
         End Function
 
-        Public Overrides Function ShouldTriggerCompletion(text As SourceText, position As Integer, trigger As CompletionTrigger, Optional roles As ImmutableHashSet(Of String) = Nothing, Optional options As OptionSet = Nothing) As Boolean
-            options = If(options, _workspace.Options)
-
-            If Not options.GetOption(CompletionOptions.TriggerOnTyping, Me.Language) Then
-                Return False
-            End If
-
-            If trigger.Kind = CompletionTriggerKind.Deletion AndAlso (Char.IsLetterOrDigit(trigger.Character) OrElse trigger.Character = "."c) Then
-                Return True
-            Else
-                Return MyBase.ShouldTriggerCompletion(text, position, trigger, roles, options)
-            End If
+        Friend Overrides Function SupportsTriggerOnDeletion(options As OptionSet) As Boolean
+            ' If the option is null (i.e. default) or 'true', then we want to trigger completion.
+            ' Only if the option is false do we not want to trigger.
+            Dim opt = options.GetOption(CompletionOptions.TriggerOnDeletion, Me.Language)
+            Return If(opt = False, False, True)
```

Comment:
Since you handle the default values in the `TryFetch` it seems like you could just have this option be a bool and not have to handle it here...